File: GolangSolutions/String/M1678_设计Goal解析器.go
package String

import "strings"

func interpret(cmd string) string {
	res := &strings.Builder{}
	for i, c := range cmd {
		if c == 'G' {
			res.WriteByte('G')
		} else if c == '(' {
			if cmd[i+1] == 'a' {
				res.WriteString("al")
			} else {
				res.WriteByte('o')
			}
		}
	}

	return res.String()
}

File: CPPSolutions/LeetCode/String/M1678_设计Goal解析器.cpp
#include <bits\stdc++.h>

using namespace std;

class Solution {
public:
    string interpret(string cmd) {
        string res;
        for (int i = 0; i < cmd.size(); i++) {
            if (cmd[i] == 'G') {
                res += "G";
            } else if (cmd[i] == '(') {
                if (cmd[i + 1] == 'a') {
                    res += "al";
                } else {
                    res += "o";
                }
            }
        }

        return res;
    }
};